tests/unit/test_client.py

```
@@ -0,0 +1,357 @@
+    
+    def test_core_client_performance(self, benchmark, core_client, sample_html):
+        """Benchmark CoreClient HTML processing performance."""
+        result = benchmark(core_client.process_html, sample_html)
+        assert result is not None
+        
+    @patch('requests.get')
+    def test_network_client_performance(self, benchmark, mock_get, network_client, mock_response):
+        """Benchmark NetworkClient URL fetching performance."""
+        mock_get.return_value = mock_response
+        result = benchmark(network_client.fetch_url, "https://example.com")
+        assert result is not None
```

Ruff (0.11.9)

No newline at end of file

Add trailing newline

(W292)